spec/system/moderation/users_spec.rb

```diff
@@ -94,4 +94,28 @@
       expect(page).to have_content "Hidden"
     end
   end
+
+  scenario "Block a user removes all their roles" do
+    admin = create(:administrator).user
+    user = create(:user, username: "Budget administrator")
+    budget = create(:budget, administrators: [create(:administrator, user: user)])
+    debate = create(:debate, author: user)
+    login_as(admin)
+    visit admin_budget_budget_investments_path(budget)
+
+    expect(page).to have_select options: ["All administrators", "Budget administrator"]
+
+    login_as(create(:moderator).user)
```

Comment:
@Senen We're creating a record after starting the browser, which might cause concurrency issues in the database. Maybe we can create the record before starting the browser, or perhaps we can remove this line and block the user while being logged in as an admin :thinking:.

```suggestion
```

Reply:
If we remove this line, we can also remove the line `login_as(admin)` which appears later in the test.

Reply:
Without those line works fine 👍🏼